packages/gestalt/src/Badge.tsx

```diff
@@ -71,52 +82,123 @@ export default function Badge({ position = 'middle', text, type = 'info', toolti
     'warning': 'workflow-status-warning',
     'success': 'check-circle',
     'neutral': 'lock',
-    'recommendation': 'performance-plus',
+    'recommendation': 'sparkle',
     'darkWash': 'info-circle',
     'lightWash': 'info-circle',
   });
 
+  const COLOR_ICON_MAP = Object.freeze({
+    'info': 'info',
+    'error': 'error',
+    'warning': 'warning',
+    'success': 'success',
+    'neutral': isInVRExperiment ? 'default' : 'inverse',
+    'recommendation': 'recommendation',
+    'darkWash': 'light',
+    'lightWash': 'dark',
+  });
+
+  const COLOR_TEXT_MAP = Object.freeze({
+    'info': 'shopping',
+    'error': 'error',
+    'warning': 'warning',
+    'success': 'success',
+    'neutral': isInVRExperiment ? 'default' : 'inverse',
+    'recommendation': 'recommendation',
+    'darkWash': 'light',
+    'lightWash': 'dark',
+  });
+
   let styleType: TypeOptions | InteractiveTypeOptions = type;
 
   if (shouldUseTooltip) {
     styleType = `interactive-${type}`;
   }
 
-  const csBadge = cx(styles.Badge, styles[position], styles[styleType]);
+  const { handleOnBlur, handleOnFocus, handleOnMouseEnter, handleOnMouseLeave, isFocused } =
+    useInteractiveStates();
+
+  const cxStyles = useMemo(
+    () =>
+      cx(styles.badge, styles[styleType], {
+        [styles.middle]: !shouldUseTooltip && position === 'middle',
```

Comment:
Does this have to be in both places? `positionStyles` and `cxStyles`

Reply:
yep